### PR TITLE
View post version history with differences

### LINKED ISSUE #64: View post version history with differences
**Problem**
The Gigs board currently does not display the editing history of posts, even though the information is stored in a smart contract. This makes it difficult for users to understand how the post has evolved over time, track the changes, and verify the accuracy.

**User Story**
As a Gigs board user, I want to view the history of a post and see the differences between each version so that I can better understand how the post changed over time.

**Acceptance Criteria**
- The version history is accessible from each post
- The version history displays a timestamp of each version and the the corresponding user account 
- The version history shows the differences between each version, highlighting any additions or deletions
- The version history is user friendly & intuitive
- Optional: The version history allows the user to compare any two versions

**Note**
- This idea was featured as a bounty at the [ETH Denver Hackathon](https://www.ethdenver.com). See proposed https://github.com/near/devgigsboard-widgets/pull/62.

### PR DESCRIPTION
**Pull request by petersalomonsen**
**NOTE**: The preview URL below now contains the deployment of https://github.com/near/neardevhub-widgets/pull/197
I will change back to the contents of this PR once that other is completed.

Preview URL: https://near.org/devgovgigs.petersalomonsen.near/widget/gigs-board.pages.Post?id=8&timestamp=1688009948396559326&compareTimestamp=1688009918129563930

image

image

Based on the work here: https://near.social/#/markeljan.near/widget/PostWithHistory?id=268&timestamp=1682382435187494750&compareTimestamp=1677857800212765894&referral=

resolves #64

### CODE REVIEW

**Comment by ailisp** [comment]
I can't trigger the post history from your preview by clicking on the clock icon:

image

Did you override the preview with the other branch?

**Comment by petersalomonsen** [comment]
yes I did override the other branch. I've rebased on top of main now, and the preview url is working now, but I see now that there's an issue with post editing preview after rebase, so I need to look into that..

I've changed back to draft, and let you know as soon as I've fixed it.

**Comment by petersalomonsen** [comment]
Issue with draft previews and history should be fixed now. Also added some more spacing on post confirmation as mentioned here ( https://github.com/near/neardevhub-widgets/pull/197#pullrequestreview-1520995410 )

File: src/gigs-board/entity/post/Post.jsx [hunk 1]
```diff
@@ -651,16 +672,95 @@ const descriptionArea = isUnderPost ? (
...
+
+          <Widget
+            src="markeljan.near/widget/MarkdownDiff"
+            props={{
+              post: post,
+              currentCode: combineText(
+                swapTimestamps ? compareSnapshot : snapshot
+              ),
+              prevCode: combineText(
+                swapTimestamps ? snapshot : compareSnapshot
```
Comment (ailisp) [inline src/gigs-board/entity/post/Post.jsx:751]: Nice normalization!

**Review by ailisp — APPROVED** [review]
Looks great! Thank you

**Pull request merged**